Grafana-prometheus

Kether180 opened the pull request:
It is working but still needs to be run it on digital ocean.

I would like your thoughts and also to know how can I add it to digital ocean.

Kether180 commented:
Prometheus:

http://localhost:9090/

http://localhost:9090/metrics

Grafana :

http://localhost:3000/

MikkelNilsson commented on minitwit-backend/Controllers/SimulatorController.cs:
```diff
@@ -268,7 +276,12 @@ public async Task<ActionResult<List<User>>> GetUserFollowers(string username, in
+            return Ok(new FollowsRes() { follows = followingRes });
+        }
+
+        internal static void HttpClientName(IServiceProvider arg1, HttpClient arg2)
+        {
+            throw new NotImplementedException();
```
What does this do? is it something that Prometheus thing? I'm just curious because it just throws a NotImplementedException

Kether180 replied:
Yes, it is part of Prometheus, but I found a way of implementing it through other way in c # .

I was trying to implement it something else but we don't need it , forgot to delete it.

ChristofferTofteng commented:
Now i will admit i don't entirely know how prometheus works with C# (or even really python as the lecture is a bit ago) but on his slides he has an endpoint called `/metrics` for Prometheus to scrape, this seems to be missing here.
There also seems to be missing a `prometheus.yml` file.

Kether180 commented:
@ChristofferTofteng the prometheus.yml file is there for sure .

In c# there is another way of adding the /metrics endpoint , check the files simulatorController and also Program.cs . There are also c# libraries installed that are doing the job. Lil bit different than in Python.

Kether180 commented:
@ChristofferTofteng  you were right, just checked.

MikkelNilsson commented on docker-compose.yml:
```diff
@@ -6,20 +6,46 @@ services:
        POSTGRES_PASSWORD: dbuserpassword
      ports:
-      - "5432:5432"
+      - "5433:5432"
```
This is not correct, im pretty sure of it

MikkelNilsson replied:
It should be kept the same `- "5432:5432"`

Kether180 replied:
done. :)

MikkelNilsson commented:
And again, please make sure to add auto-generated files to .gitignore (.vscode folder)

Kether180 commented:
For sure .

Kether180 commented:
added .gitignore (.vscode folder) . @MikkelNilsson 

Could you please check the conflicts so we can merge it to dev branch ? I've checked the development branch and the differences are that in docker-compose  file you will find  ports and data to use grafana and prometheus, and in 

The simulatorcontroller file is just showing what I added and the rest is just that I erase the throwException that I had.

Kether180 commented:
Just fixed the conflicts and ready to Merge:) !
